app.js

```diff
@@ -2,10 +2,26 @@ const express = require("express");
 const app = express();
 app.use(express.json());
 const getAllTopics = require("./controllers/topics.controller");
-const listAll = require("./controllers/endpoints.controller");
+const listAllEndpoints = require("./controllers/endpoints.controller");
+const listArticlesById = require("./controllers/articles.controller");
 
 app.get("/api/topics", getAllTopics);
 
-app.get("/api/endpoints", listAll);
+app.get("/api/endpoints", listAllEndpoints);
 
-module.exports = app;
\ No newline at end of file
+app.get("/api/articles/:article_id", listArticlesById);
+
+app.use((err, req, res, next) => {
+  console.log("Error In Middleware", err);
+  if (err.code === "22P020" || err.code === "23502") {
+    res.status(400).send({ msg: "Bad Request" });
+  } else if (err.code === "23503") {
+    res.status(404).send({ msg: "Not Found" });
+  } else if (err.status === 404) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
```

Comment:
I would suggest separating out your error handling middleware into a separate file at this point, as it's becoming very big in this app :)